src/include/session.h

```diff
@@ -66,6 +66,7 @@ struct __wt_session_impl {
     uint64_t cache_wait_us;        /* Wait time for cache for current operation */
     uint64_t operation_start_us;   /* Operation start */
     uint64_t operation_timeout_us; /* Maximum operation period before rollback */
+    u_int api_call_counter;        /* Depth of api calls */
```

Comment:
Are you confident that this will always start at 0?  Could we add some checking in to make sure that the reference counting is right? i.e: ensuring it is 0 when we know it should be in that state?

Reply:
We use calloc to alloc the whole session array, so it is initialized to 0:

```
    /* WT_SESSION_IMPL array. */
    WT_RET(__wt_calloc(session, conn->session_size, sizeof(WT_SESSION_IMPL), &conn->sessions));
```

Reply:
I'll think about how to check that.

Reply:
I have a bad idea, I think we may be able to cross reference that with s->name to make sure the reference counting is correct.